Assembler - Fastrack Next Assembler Tasks Jun 2023

miksansegundo opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #78513

## Proposed Changes

### Large preview placeholder updates

- Remove icon
- Update copy
- Remove orphan (a single word on the last line of text)
- Remove `Add header` button from the colors and fonts placeholder and update the title copy

**Regular placeholder after removing the icon**

<img width="657" alt="Screenshot 2566-06-26 at 14 22 46" src="https://github.com/Automattic/wp-calypso/assets/1881481/c0da3f85-f109-41b8-b031-8b5d79128e10">

**Colors and fonts placeholder**
 
|Before|After|
|---|---|
|<img width="653" alt="Screenshot 2566-06-27 at 17 05 01" src="https://github.com/Automattic/wp-calypso/assets/1881481/398ef251-856b-409d-986a-5316731f5e71">|<img width="709" alt="Screenshot 2566-06-27 at 17 05 23" src="https://github.com/Automattic/wp-calypso/assets/1881481/425a0c61-fbca-4817-aa9a-a44524ccb4e8">|


**More copy updates**

- Update title and description copy on main screen
- Update title of Header and Footer 
- Update Continue button label and remove the hint
- Update copy to use sections and patterns
- Update description copy on categories screen


**More orphans removed**

- Remove assembler CTA orphan

<img width="966" alt="Screenshot 2566-06-26 at 14 22 29" src="https://github.com/Automattic/wp-calypso/assets/1881481/4df6bd85-20dd-44d2-8122-63f47f1e32f2">

<img width="772" alt="Screenshot 2566-06-26 at 14 39 39" src="https://github.com/Automattic/wp-calypso/assets/1881481/aeed195a-b8d4-4b58-8de5-d3ddd08b72de">


- Remove large preview placeholder orphan
- Remove survey orphan
- Remove screen description orphans

|Main|Add header|Sections|Add sections|Add footer|Colors|Fonts|
|--|--|--|--|--|--|--|
|<img width="358" alt="Screenshot 2566-06-27 at 16 50 49" src="https://github.com/Automattic/wp-calypso/assets/1881481/9ca877b5-1dbc-445b-9da3-76d7d17506b1">|<img width="359" alt="Screenshot 2566-06-27 at 16 51 42" src="https://github.com/Automattic/wp-calypso/assets/1881481/b5c70070-17ea-4d47-a3c8-6919a58893b6">|<img width="357" alt="Screenshot 2566-06-26 at 14 32 35" src="https://github.com/Automattic/wp-calypso/assets/1881481/6a284d07-4922-4fb5-bbc0-6ed92f8c6f21">|<img width="359" alt="Screenshot 2566-06-26 at 14 32 09" src="https://github.com/Automattic/wp-calypso/assets/1881481/6096519d-4260-4b69-a0b3-c97e463b5cb7">|<img width="358" alt="Screenshot 2566-06-27 at 16 51 56" src="https://github.com/Automattic/wp-calypso/assets/1881481/d9223d09-f81c-4945-9c43-67bbe9ebc75a">|<img width="359" alt="Screenshot 2566-06-26 at 14 31 49" src="https://github.com/Automattic/wp-calypso/assets/1881481/25e238c3-a057-4d39-b95b-c4eaba9bfe33">|<img width="358" alt="Screenshot 2566-06-26 at 14 38 44" src="https://github.com/Automattic/wp-calypso/assets/1881481/9a8f8f5c-b31f-4340-a674-1d9f0674ca3c">|


**This PR doesn't fix sidebar alignment issues. There is another [PR](https://github.com/Automattic/wp-calypso/pull/78587) only for that.**

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->
* Using the Calypso Live link below 
* Access the assembler `/setup/with-theme-assembler/patternAssembler?siteSlug={ YOUR SITE }&theme=blank-canvas-3`
* Test the whole experience from the CTA until you land on the editor
* Verify the copy makes sense and there aren't typos
* Verify the icon has been removed from the large placeholder 
* Test the copy twice, on a window size larger and smaller than 1280px to verify there aren't orphans in the copy when the sidebar width is reduced

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

miksansegundo committed:
Remove icon from preview placeholder

miksansegundo committed:
Update copy on preview placeholder

miksansegundo committed:
Remove assembler CTA widow

miksansegundo committed:
Update title and description copy on main screen

miksansegundo committed:
Update Continue button copy

miksansegundo committed:
Update copy to use sections and patterns

miksansegundo committed:
Use main description for virtual themes too

miksansegundo committed:
Update copy description on categories screen

miksansegundo committed:
Fix preview placeholder orphan

miksansegundo committed:
Remove cta orphan

miksansegundo committed:
Remove survey orphan

miksansegundo committed:
Remove screen description orphans

miksansegundo committed:
Update continue button copy

arthur791004 reviewed (approved):
LGTM 👍

But there are lots of copy changes, so we're not able to ship it immediately. Instead, we have to add the `String Freeze` label and wait for the translations

taipeicoder reviewed (approved):
I have two feedbacks:

1. The label should be Styles and not Style:

| Current | Expected |
| --- | --- |
| ![Screenshot 2023-06-26 at 3 14 24 PM](https://github.com/Automattic/wp-calypso/assets/797888/37775113-303f-42b4-a7df-0d4965e3ff21) | ![Screenshot 2023-06-26 at 3 09 52 PM](https://github.com/Automattic/wp-calypso/assets/797888/cc804c4d-6242-4533-9851-81350f295939) |

2. The distance between the description text and the nav items should be 32px:
![Screenshot 2023-06-26 at 2 57 17 PM](https://github.com/Automattic/wp-calypso/assets/797888/00eaecb0-b1e3-46e2-b67a-de0d9577b998)

Also, the unit test snapshot needs to be updated since the CTA markup also changed.

miksansegundo commented:
@taipeicoder the alignment issues are fixed in https://github.com/Automattic/wp-calypso/pull/78587

miksansegundo committed:
Update styles copy

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8065692

Hi @miksansegundo, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Welcome to your homepage.`
- `Find the right patterns for your homepage by exploring the pattern categories below.`
- `Discover your ideal color blend, from free to premium styles.`
- `Elevate your design with expertly curated font pairings, including free and premium.`
- `Pick the footer appears at the bottom of every page and shows useful links and your contact information.`
- `Pick the header that appears at the top of every page and shows your site logo, title and navigation.`
- `Customize everything in your homepage by adding patterns and choosing styles.`
- `Design your own homepage`
- `Patterns`
- `Ready to keep editing?`
- `Save sections`
- `Can’t find something you like?`
- `Jump right into the editor to design your homepage from scratch.`

Thank you in advance!

miksansegundo committed:
Update snap test

fushar commented on client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-footer.tsx:
```diff
@@ -33,7 +33,7 @@ const ScreenFooter = ( {
 				title={ translate( 'Footer' ) }
 				description={ translate(
-					'The footer appears at the bottom of a site and shows useful links and contact information.'
+					'Pick the footer appears at the bottom of every page and shows useful links and your contact information.'
```
```suggestion
					'Pick the footer that appears at the bottom of every page and shows useful links and your contact information.'
```

taipeicoder commented on client/components/themes-list/test/__snapshots__/index.jsx.snap:
```diff
@@ -31,7 +31,9 @@ exports[`ThemesList should render a div with a className of "themes-list" 1`] =
         class="pattern-assembler-cta__subtitle"
       >
-        Can't find something you like? Jump right into the editor to design your homepage from scratch.
+        Can’t find something you like? 
```
The snapshots are very sensitive to whitespace differences, so make sure they are identical!

fushar replied:
While we're at it -- does it matter if we use `'` or `’`? I can't find any resources / guidelines on that for Automattic.

miksansegundo replied:
Here is a reference puPv3-6m9-p2#comment-19137

> ...to pay more attention to proper quote marks and apostrophes rather than using ' and "

fushar replied:
TIL thanks!

miksansegundo committed:
Fix footer copy

miksansegundo committed:
Fix snap test whitespace

miksansegundo committed:
Fix snap test whitespace

autumnfjeld commented:
@miksansegundo  If it is easy to make the following update, please do. Otherwise we'll wait for another time.  **Not a deal breaker**.   I ask because it isn't very smooth English, it doesn't sound good. 

![image](https://github.com/Automattic/wp-calypso/assets/5665959/1289787f-9464-49ac-91ad-8c8ba203d5fa)

Change to:
" Add and arrange patterns to make your homepage unique and showcase the content of your site."


But again, no worries if it is too rushed. We can do another copy adjustment after Lucas is back, as we know there are still copy updates to sort out in his latest Figma.

miksansegundo committed:
Update sections description copy

miksansegundo commented:
@autumnfjeld done ✅, looks good on both breakpoints 👍

<img width="358" alt="Screenshot 2566-06-27 at 06 42 36" src="https://github.com/Automattic/wp-calypso/assets/1881481/e03c50ce-8ff4-404e-a956-092f287ba955">

<img width="313" alt="Screenshot 2566-06-27 at 06 42 26" src="https://github.com/Automattic/wp-calypso/assets/1881481/c0b0ca88-21ee-427f-93f0-77de79fddae5">

okmttdhr reviewed (approved):
❤️ 
LGTM

miksansegundo committed:
Update copy after review call

miksansegundo committed:
Remove orphans

miksansegundo commented:
I added the following tasks and updated the screenshots in the description:

- ⭐ “Add” - using it consistently, added to Header & Footer
- ⭐ Delete the “Ready to keep editing?”
- ⭐ When clicking on color/font
    - Remove the button
    - Use the same title “Welcome…..”

- ⭐ “Customize everything in your homepage by first adding patterns and then choosing styles.”
  - I removed "in your homepage" to avoid an orphan because it sounds redundant, see the next screenshots:

|BEFORE|AFTER|
|--|--|
|<img width="359" alt="Screenshot 2566-06-27 at 16 49 07" src="https://github.com/Automattic/wp-calypso/assets/1881481/184fa049-15ac-4542-ad7d-b189e4d60951">|<img width="358" alt="Screenshot 2566-06-27 at 16 50 49" src="https://github.com/Automattic/wp-calypso/assets/1881481/6753dbbc-7324-469c-9cc6-407585a3dc5f">|

Does it sound good? Any better copy ideas to remove that orphan? cc: @taipeicoder @autumnfjeld

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8065692).

taipeicoder commented:
@miksansegundo we can finally ship this 🙂

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8065692).
